extensions/amp-carousel/0.1/slides.js

```diff
@@ -81,6 +93,55 @@ export class AmpSlides extends BaseCarousel {
             });
       }
     }
+    this.tryAutoplay_(1, true);
+  }
+
+  /**
+   * Sets up the `autoplay` configuration.
+   * @private
+   */
+  setupAutoplay_() {
+    if (!this.isAutoplaying_) {
+      return;
+    }
+
+    let autoplayValue = Number(this.element.getAttribute('autoplay'));
+    // If it isn't a number and is not greater than 0 then don't assign
+    // and use the default.
+    if (autoplayValue > 0) {
+      // Guard against autoplayValue that is lower than 1s to prevent
+      // people from crashing the runtime with providing very low delays.
+      this.autoplayDelay_ = Math.max(1000, autoplayValue);
+    }
+
+    // By default `autoplay` should also mean that the current carousel slide
+    // is looping. (to be able to advance past the last item)
+    if (!this.element.hasAttribute('loop')) {
+      this.element.setAttribute('loop', '');
+    }
+
+    // Setup microtask to let the runtime settle.
+    // TODO: (erwinm) experiment with only letting autoplay run if the
+    // carousel is in the viewport.
+    timer.delay(this.tryAutoplay_.bind(this, 1, true), 0);
+  }
+
+  /**
+   * Sets up the autoplay delay if necessary.
+   * @private
+   * @param {number} dir -1 or 1
+   * @param {boolean} animate
+   */
+  tryAutoplay_(dir, animate) {
+    if (!this.isAutoplaying_) {
+      return;
+    }
+
+    if (this.autoplayTimeoutId_ !== null) {
+      timer.cancel(this.autoplayTimeoutId_);
+    }
+    this.autoplayTimeoutId_ = timer.delay(this.go.bind(this, dir, animate),
```

Comment:
Yeah, we can just continue autoplay for now. I think this is easy to address later.